aioshelly/rpc_device/device.py

```diff
@@ -538,3 +544,29 @@ def _parse_dynamic_components(self, components: dict[str, Any]) -> None:
                 for item in self._dynamic_components
             }
         )
+
+    async def _retrieve_blutrv_components(self, components: dict[str, Any]) -> None:
+        """Retrieve BLU TRV components."""
+        if self.model != MODEL_BLU_GATEWAY_GEN3:
+            return
+
+        if TYPE_CHECKING:
+            assert self._config
+            assert self._status
```

Comment:
Since we can call this method when periodically pulling a device and we had history of race conditions of initializing the device multiple times, I think we should explicitly check that these values are not none and log error if they are.